src/controller/CommissioningDelegate.h

```diff
@@ -573,6 +576,18 @@ class CommissioningParameters
     }
     void ClearICDStayActiveDurationMsec() { mICDStayActiveDurationMsec.ClearValue(); }
 
+    Span<const app::AttributePathParams> GetExtraReadPaths() const { return mExtraReadPaths; }
+
+    // Additional attribute paths to read as part of the kReadCommissioningInfo stage.
+    // These values read from the device will be available in ReadCommissioningInfo.attributes.
+    // Clients should avoid requesting paths that are already read internally by the commissioner
+    // as no consolidation of internally read and extra paths provided here will be performed.
+    CommissioningParameters & SetExtraReadPaths(Span<const app::AttributePathParams> paths)
+    {
+        mExtraReadPaths = paths;
```

Comment:
Needs to be added to ClearExternalBufferDependentValues.

And AutoCommissioner should be providing storage for a copy of this thing...  Except AttributePathParams can also hold pointers to external buffers, so we really need to think about ownership issues here.